hack/generator/cmd/gen/gen.go

```diff
@@ -29,103 +29,18 @@ func NewGenCommand() (*cobra.Command, error) {
 		Short: "generate K8s infrastructure resources from Azure deployment template schema",
 		Run: xcobra.RunWithCtx(func(ctx context.Context, cmd *cobra.Command, args []string) error {
 
-			//TODO extract into a new type (maybe 'Generator') so that this file can concentrate on command line processing
-			configuration, err := loadConfiguration("azure-cloud.yaml")
+			cg, err := codegen.NewCodeGenerator("azure-cloud.yaml")
 			if err != nil {
-				log.Printf("Error loading configuration: %v\n", err)
-				writeSampleConfig("public-cloud-sample.yaml")
+				klog.Errorf("Error loading configuration: %v\n", err)
 				return err
 			}
 
-			err = configuration.Validate()
-			if err != nil {
-				log.Printf("Configuration invalid: %v\n", err)
-				return err
-			}
-
-			log.Printf("Loading schema %s", configuration.SchemaURL)
-			schema, err := loadSchema(configuration.SchemaURL)
-			if err != nil {
-				log.Printf("Failed to load schema: %v\n", err)
+			err = cg.Generate(ctx, "apis")
+			if err!=nil {
```

Comment:
minor: formatting, I believe you need spaces between the `err!=nil` bits -- CI will likely complain about this now but may not have before